### ISSUE TITLE
Week 5 Assignment Comments

### ISSUE THREAD

**Issue by mhogaz** [gge-ucd/r-davis-in-class-brcurrinder#2]

**Comment by mhogaz** [comment]
Hello Bryan! Your code was quite good. Very well annotated and easy to understand and run. 

The only problem I ran into was that for the surveys_avg_weight and both challenges, you piped "view" onto the end. Two problems occurred: one, you didn't capitalize the "V", so the entire chunk of code didn't run and two, by adding view onto the end it caused the entire tibble you had just created to become a value (I think? It was under the values section in Environment instead of the data section) of class NULL and then could not be used in any subsequent code because the tibble did not exist. I don't really know why, maybe @marthawohlfeil, @mcmaurer, or @ryanpeek could help us out?

Regardless, when I ran the code without View it was flawless :)

**Comment by ryanpeek** [comment]
Great question @mhogaz! And good job @brcurrinder!

There are ways to get around this and be able to use both the `View()` option, and the `%>%` operator...it's something we didn't cover in class but may be worth mentioning at the start of class tomorrow. There are a few other "pipe" options that are part of the [`magrittr`](https://cran.r-project.org/web/packages/magrittr/vignettes/magrittr.html) package (which is what the `tidyverse` is using behind the scenes). I'll mention one in particular which is kind of cool and useful:

- The "Tee" pipe, or **`%T>%`**: which allows you to basically have something that is a split or additional T from your regular pipe. So for the example you mentioned @mhogaz, it would look like the chunk below. If you run this, you'll notice it both outputs the View() as requested, but it *also* assigns the tibble object to the environment called `surveys_avg_weight` which is available for future use if need be. The only difference is the `%T>%`.

```
surveys_avg_weight <- surveys %>% 
  filter(!is.na(weight)) %>% #filter out NAs
  group_by(species_id, sex) %>% #group by species+sex combo
  mutate(mean_weight = mean(weight)) %>% #add new column of mean weight
  select(species_id, sex, weight, mean_weight) %T>% #select only these columns
  View()
```
Basically anything that comes after the **`%T>%`** will take the left hand side, and send it to a separate output or component, that diverges or tees from the original call to assign all the stuff on the left hand side to the `surveys_avg_weight`. These can appear anywhere in a pipe, but they can be a little tricky if you use them in the middle of a chain of pipes. Happy to chat more about this tomorrow.

Hope this is helpful! 👍
